cohydra/channel/wifi.py

```diff
@@ -261,3 +279,25 @@ def prepare(self, simulation):
         for interface in self.interfaces:
             pcap_log_path = os.path.join(simulation.log_directory, interface.pcap_file_name)
             self.wifi_phy_helper.EnablePcap(pcap_log_path, interface.ns3_device, True, True)
+
+    def set_delay(self, delay):
+        logger.info(f'Set delay of channel {self.channel_name} to {delay}')
+
+        if re.match("^\d+ms", delay):
+            self.delay = delay
+        elif re.match("^\d+s", delay):
+            self.delay = str(int(delay[:-1])*1000)+"ms"
```

Comment:
Nit: regex is not terminated with `$` -- this would also match '10s foobar'. Even more elegant than the substring operator would be to extract the numeric value from the regex match...

Also you're mixing quotation mark styles (`"`/`'`) here and in the above logger statement. Personally, I also prefer spaces around arithmetic operators for better legibility.